flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1705,6 +1710,64 @@ mlir::Value IntrinsicLibrary::genConjg(mlir::Type resultType,
       cplx, negImag, /*isImagPart=*/true);
 }
 
+// COUNT
+fir::ExtendedValue
+IntrinsicLibrary::genCount(mlir::Type resultType,
+                           llvm::ArrayRef<fir::ExtendedValue> args) {
+  assert(args.size() == 3);
+
+  // Handle mask argument
+  auto mask = builder.createBox(loc, args[0]);
+  fir::BoxValue maskTmp = mask;
```

Comment:
Can the above two lines be combined into 1? You can use `fir::getBase(mask)` to return the `Value` from the `BoxValue`, I believe. (It seems a little odd that `createBox` doesn't already return a `BoxValue`.)